### PR TITLE
Add additional global defintions

### PR DESCRIPTION
**Pull request by ksonney**
This adds a global specifier to both us government providers, as well as a top-level "global" provider.

### CODE REVIEW

**Comment by ksonney** [comment]
This is in reference to https://github.com/elastic/cloud/issues/71813

File: pkg/cso/v1/validator.go [hunk 1]
```diff
@@ -116,6 +116,7 @@ var cloudProviderRegions = map[string]types.StringArray{
 		"sa-east-1",
 	},
 	"aws-us-gov": {
+		"global",
```
Comment (Zenithar) [inline pkg/cso/v1/validator.go:119]: Ok.

File: pkg/cso/v1/validator.go [hunk 2]
```diff
@@ -187,11 +188,15 @@ var cloudProviderRegions = map[string]types.StringArray{
 		"brazilsoutheast",
 	},
 	"azure-us-gov": {
+		"global",
 		"usgovvirginia",
 		"usgoviowa",
 		"usgovarizona",
 		"usgovtexas",
 	},
+	"global": {
```
Comment (Zenithar) [inline pkg/cso/v1/validator.go:197]: You will have a secret addressed by `infra/global/<account>/global/<service>/<suffix>` path which can be interpreted as provider / region unbound secret, which is the definition of a platform secret in CSO.

From a security point of view, having a shared secret across all infra is not correct. 

As a reminder :

`infra` ⇒ Infrastructure secrets tightly coupled to infrastructure provider

Capture d’écran 2020-12-17 à 07 28 42

`platform` ⇒ Logical infrastructure (spread over multiple infra)

Capture d’écran 2020-12-17 à 07 29 13

I suggest you to migrate this secret in the platform ring.

`platform/production/<platform-name>/global/<component>/<suffix>`

Visually the infra / platform distribution could be drawn like the following diagram :

CSO_RING_1_2

And Platform / Product / App relations

CSO_3_4



CC @elastic/cloud-security

Comment (ksonney) [reply pkg/cso/v1/validator.go:197]: I think I get it, let me make the proper adjustments.

**Comment by ksonney** [comment]
git rebase is up in my grill again, so standby while I fix that.

File: pkg/cso/v1/validator.go (hunk 2 above)
Comment (ksonney) [reply pkg/cso/v1/validator.go]: OK, removed `infra/global` in latest changeset.

**Comment by Zenithar** [comment]
Changes already merged in 0.1.6.

**Pull request closed without merging**